starter_code/src/movies.js

```diff
@@ -1,15 +1,373 @@
+    year: 2002,
+    director: "Peter Jackson",
+    duration: "2h 59min",
+    genre: ["Adventure", "Drama", "Fantasy"],
+    rate: 8.7
+  },
+  {
+    title: 'One Flew Over the Cuckoo"s Nest',
+    year: 1975,
+    director: "Milos Forman",
+    duration: "2h 13min",
+    genre: ["Drama"],
+    rate: 8.7
+  },
+  {
+    title: "Goodfellas",
+    year: 1990,
+    director: "Martin Scorsese",
+    duration: "2h 26min",
+    genre: ["Crime", "Drama"],
+    rate: 8.7
+  },
+  {
+    title: "The Matrix",
+    year: 1999,
+    director: "Lana Wachowski",
+    duration: "2h 16min",
+    genre: ["Action", "Sci-Fi"],
+    rate: 8.7
+  },
+  {
+    title: "Shichinin no samurai",
+    year: 1954,
+    director: "Akira Kurosawa",
+    duration: "3h 27min",
+    genre: ["Adventure", "Drama"],
+    rate: 8.7
+  },
+  {
+    title: "Star Wars",
+    year: 1977,
+    director: "George Lucas",
+    duration: "2h 1min",
+    genre: ["Action", "Adventure", "Fantasy", "Sci-Fi"],
+    rate: 8.7
+  },
+  {
+    title: "Cidade de Deus",
+    year: 2002,
+    director: "Fernando Meirelles",
+    duration: "2h 10min",
+    genre: ["Crime", "Drama"],
+    rate: 8.7
+  },
+  {
+    title: "Se7en",
+    year: 1995,
+    director: "David Fincher",
+    duration: "2h 7min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: "The Silence of the Lambs",
+    year: 1991,
+    director: "Jonathan Demme",
+    duration: "1h 58min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: 'It"s a Wonderful Life',
+    year: 1946,
+    director: "Frank Capra",
+    duration: "2h 10min",
+    genre: ["Drama", "Family", "Fantasy"],
+    rate: 8.6
+  },
+  {
+    title: "La vita è bella",
+    year: 1997,
+    director: "Roberto Benigni",
+    duration: "1h 56min",
+    genre: ["Comedy", "Drama", "War"],
+    rate: 8.6
+  },
+  {
+    title: "The Usual Suspects",
+    year: 1995,
+    director: "Bryan Singer",
+    duration: "1h 46min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.6
+  },
+  {
+    title: "Léon",
+    year: 1988,
+    director: "Luc Besson",
+    duration: "1h 50min",
+    genre: ["Crime", "Drama", "Thriller"],
+    rate: 8.6
+  }
+];
+
+console.log("----------------------------");
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(arr) {
+  if (arr.length == 0) {
+    return [null];
+  } else {
+    arr.sort(function(a, b) {
+      return a.year == b.year ? (a.title > b.title) - 0.5 : a.year - b.year;
+    });
+    return arr;
+  }
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+  let arr2 = [];
+  let arr3 = [];
+  if (arr.length === 0) {
+    return 0;
+  } else {
+    arr2 = arr.filter(a => a.director == "Steven Spielberg");
+    arr3 = arr2.filter(a => a.genre.indexOf("Drama") !== -1);
+    return arr3.length;
+  }
+}
+
```

Comment:
Here you could do everything within a single filter,  since you have 2 conditions that have to be met, you can use the && operator instead of filtering your array twice, and if nothing meets the condifitions, it will create an empty array.

Solution:

```
function howManyMovies(arr) {
 return arr.filter(m => m.director === "Steven Spielberg" && m.genre.includes("Drama")).length;
}
```